# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/tests/mochitest/l10n_mutations/test_disconnectedRoot_webcomponent.html
# Commit: 595529cd906f
# Full Hash: 595529cd906f94cdb2bc12a7a40072db0c07198e
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-03 22:01:19
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 9f3e2963d925 -r 595529cd906f dom/l10n/tests/mochitest/l10n_mutations/test_disconnectedRoot_webcomponent.html
--- a/dom/l10n/tests/mochitest/l10n_mutations/test_disconnectedRoot_webcomponent.html	Tue Aug 03 05:52:03 2021 +0000
+++ b/dom/l10n/tests/mochitest/l10n_mutations/test_disconnectedRoot_webcomponent.html	Tue Aug 03 05:52:03 2021 +0000
@@ -7,21 +7,23 @@
   <link rel="stylesheet" type="text/css" href="chrome://mochikit/content/tests/SimpleTest/test.css">
   <script type="application/javascript">
   "use strict";
-  async function* generateBundles(resourceIds) {
-    const bundle = new FluentBundle("en-US");
-    bundle.addResource(new FluentResource(`
+  const l10nReg = new L10nRegistry();
+  const fs = [
+    { path: "/localization/en-US/mock.ftl", source: `
 key1 = Key 1
 key2 = Key 2
 key3 = Key 3
 key4 = Key 4
-    `));
-    yield bundle;
-  }
+` },
+  ];
+  const source = L10nFileSource.createMock("test", ["en-US"], "/localization/{locale}", fs);
+  l10nReg.registerSources([source]);
 
   document.domLoc = new DOMLocalization(
-    [],
+    ["/mock.ftl"],
     false,
-    { generateBundles }
+    l10nReg,
+    ["en-US"],
   );
   document.domLoc.connectRoot(document.documentElement);
   </script>